aten/src/THC/generic/THCTensor.cpp

```diff
@@ -161,7 +161,9 @@ THCTensor *THCTensor_(newWithSize4d)(THCState *state, int64_t size0, int64_t siz
 THCTensor *THCTensor_(newClone)(THCState *state, THCTensor *self)
 {
   THCTensor *tensor = THCTensor_(new)(state);
-  THCTensor_(resizeAs)(state, tensor, self);
+  at::Tensor tensor_wrap = THTensor_wrap(tensor);
```

Comment:
i see that CPU version does wrapping, was it a bug that CUDA wasn't doing it?

Reply:
Probably just partial porting, as https://github.com/pytorch/pytorch/blob/320c38555eaccb0abba8fa1206e677011c15d43b/aten/src/THC/generic/THCTensorCopy.cu#L5 does wrapping. They will eventually go away as we deprecate all call sites.

Reply:
I'm a bit confused what the strategy is here.

You don't just dispatch to at::resize_as in THCTensor_(resizeAs), but you call at::resize_as here.

Reply:
Let me follow-up with two more PRs, completly removing `TH[C]Tensor_(resizeAs)` and `TH[C]Tensor_(newClone)` code. I just afraid having hundreds line code update here, will be hard to review.

Reply:
ya that's totally fine, I just wanted to understand the strategy.